sonar-application/src/main/java/org/sonar/application/Cluster.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Manager for the cluster communication between Main Processes
+ */
+public class Cluster implements AutoCloseable {
```

Comment:
Hazelcast is trapping the SHUTDOWN signal and it's stopping correctly, so there is no explicit need to shutdown him unless we are in test with multiple executions in the same JVM.

I do not see exactly how we can use `AutoCloseable` in App since we do not want to close Hazelcast unless we are stopping SonarQube. How do you want to do this ?

So from my point of view either :
* I drop `AutoCloseable` and update the tests,
* We are keeping it but it's not useful.

So I would go to drop this ? WDYT ?